src/v/redpanda/admin_server.cc

```diff
@@ -2898,6 +2899,79 @@ void admin_server::register_transaction_routes() {
       });
 }
 
+void admin_server::register_self_test_routes() {
+    register_route<user>(
```

Comment:
In the rest of the API, the pattern is generally `user` for read methods, and `superuser` for endpoints that change something.